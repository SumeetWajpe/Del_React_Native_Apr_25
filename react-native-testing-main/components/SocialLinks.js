import React from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Linking,
  Button,
  Image,
} from "react-native";

const Icons = {
  twitter: require("../assets/twitter.png"),
  instagram: require("../assets/instagram.png"),
};

const SocialLinks = ({ type, link, label }) => {
  const onSocialLinkPress = () => {
    Linking.openURL(link);
  };
  return (
    <View style={styles.social}>
      <Image
        source={Icons[type]}
        style={styles.icon}
        accessibilityRole="image"
      />

      <Text onPress={onSocialLinkPress} style={styles.label}>
        {label}
      </Text>
      <Button title="Test !"></Button>
    </View>
  );
};

const styles = StyleSheet.create({
  icon: {
    width: 20,
    height: 20,
  },
  social: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: 10,
  },
  label: {
    marginLeft: 10,
  },
});

export default SocialLinks;
