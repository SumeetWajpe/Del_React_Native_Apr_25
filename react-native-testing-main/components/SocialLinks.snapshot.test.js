import renderer from "react-test-renderer";
import SocialLinks from "./SocialLinks";
test("Link renders correctly", () => {
  const tree = renderer
    .create(
      <SocialLinks
        type="instagram"
        label="John Doe's Instagram"
        link="https://www.instagram.com"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
