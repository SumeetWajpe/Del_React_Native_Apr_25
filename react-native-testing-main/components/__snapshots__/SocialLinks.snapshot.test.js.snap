// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 10,
    }
  }
>
  <Image
    accessibilityRole="image"
    source={
      {
        "testUri": "../../../assets/instagram.png",
      }
    }
    style={
      {
        "height": 20,
        "width": 20,
      }
    }
  />
  <Text
    onPress={[Function]}
    style={
      {
        "marginLeft": 10,
      }
    }
  >
    John Doe's Instagram
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Test !
      </Text>
    </View>
  </View>
</View>
`;
